CrossCompatibility/CrossCompatibility/Common/PowerShellVersion.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.PowerShell.CrossCompatibility
+{
+    public class PowerShellVersion
```

Comment:
Ah interesting.

Well the big addition in my case is that it allows serialization and deserialization in the JSON -- something that can't be achieved with compiler macros, since we need to rehydrate new versions in old PowerShells.

My implementation is a lot more involved though too, so it's not clear that's it's preferable everywhere.